### PR TITLE
don't hang forever and retry tcp failures

### PR DESCRIPTION
**Pull request by snorecone**

### CODE REVIEW

File: adapters/syslog/syslog.go [hunk 1]
```diff
@@ -96,13 +97,90 @@ func (a *SyslogAdapter) Stream(logstream chan *router.Message) {
 		_, err = a.conn.Write(buf)
 		if err != nil {
 			log.Println("syslog:", err)
-			if reflect.TypeOf(a.conn).String() != "*net.UDPConn" {
```
Comment (grepory) [inline adapters/syslog/syslog.go:99]: lol

File: adapters/syslog/syslog.go [hunk 2]
```diff
@@ -96,13 +97,90 @@ func (a *SyslogAdapter) Stream(logstream chan *router.Message) {
...
+				if err != nil {
+					log.Println("syslog:", err)
+					return
+				}
 			}
 		}
 	}
 }
 
+func (a *SyslogAdapter) retry(buf []byte, err error) error {
```
Comment (grepory) [inline adapters/syslog/syslog.go:114]: You may as well fix the connection leak that exists here as well and do an a.conn.Close() if we decide that a connection is dead and go to reconnect.

Comment (snorecone) [inline adapters/syslog/syslog.go:114]: n i c e   c a t c h   b b

File: adapters/syslog/syslog.go (hunk 1 above)
Comment (snorecone) [inline adapters/syslog/syslog.go:99]: really cool stuff

**Comment by grepory** [comment]
@snorecone we can close this right?

**Pull request closed without merging**